pkg/csi/types/envvars.go

```diff
@@ -31,4 +31,16 @@ const (
 
 	// EnvInClusterClientBurst is the Burst for all clients to the API server
 	EnvInClusterClientBurst = "INCLUSTER_CLIENT_BURST"
+
+	// EnvVarEndpoint specifies the CSI endpoint for CSI driver.
+	EnvVarEndpoint = "CSI_ENDPOINT"
+
+	// EnvVarMode is the name of the environment variable used to specify
+	// the service mode of the plugin. Valid values are:
+	// * controller
+	// * node
+	//
+	// Depending on the value, either controller and node service will be
+	// activated (The identity service is always activated).
+	EnvVarMode = "X_CSI_MODE"
```

Comment:
Can we avoid using `X_CSI_MODE` for non-gocsi code, as it is specific to gocsi.
https://github.com/rexray/gocsi/blob/master/envvars.go#L79

Reply:
Even if we use non-gocsi, we need a variable in yaml to determine whether to run as a controller or node plugin. Are you suggesting to rename this in yaml? Then we won't be able to use the same yaml when running with gocsi or non-gocsi.